key.go

```diff
@@ -0,0 +1,784 @@
+package cose
+
+import (
+	"crypto"
+	"crypto/ecdsa"
+	"crypto/ed25519"
+	"crypto/elliptic"
+	"errors"
+	"fmt"
+	"math/big"
+	"strconv"
+
+	cbor "github.com/fxamacker/cbor/v2"
+	"golang.org/x/exp/slices"
+)
+
+const (
+	// An inviald key_op value
+	KeyOpInvalid KeyOp = -1
```

Comment:
Curious why it is `-1` not `0`? Is this value defined in any RFC doc or IANA registry?